joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/typerelations/PointerAssignmentPass.scala

```diff
@@ -0,0 +1,143 @@
+package io.joern.x2cpg.passes.typerelations
+
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes._
+import io.shiftleft.codepropertygraph.generated.{EdgeTypes, Operators}
+import io.shiftleft.passes.ForkJoinParallelCpgPass
+import io.shiftleft.semanticcpg.language._
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+/** Creates an intraprocedural pointer assignment graph. This provides may point to information for each identifier or
+  * field identifier.
+  */
+class PointerAssignmentPass(cpg: Cpg) extends ForkJoinParallelCpgPass[Method](cpg) {
+
+  override def generateParts(): Array[Method] = cpg.method.toArray
+
+  override def runOnPart(builder: DiffGraphBuilder, method: Method): Unit = {
+    val allocSites = method.ast.isCall.name(Operators.alloc, Operators.arrayInitializer).l
```

Comment:
Same potential issue as below wrt inner methods.

You only need to check whether allocSites is nonempty, not find all of them, right? You can use headOption for that.